src/persist/src/indexed/encoding.rs

```diff
@@ -1469,6 +1493,61 @@ mod tests {
                 "next stream Id(2), but only registered 1 ids and deleted 0 ids"
             ))
         );
+
```

Comment:
Sorry for the drive-by nit: this test is already very long as it is, it might be better to create a new test function just for this uniqueness check.

Feel free to ignore.

Reply:
No need to apologize! The way this file is structured each type's `validate` function currently gets its own test. I'm going to stick to that convention for now even though I agree with you this test is quite long.